### PR TITLE
[SVLS-5049] It is okay to have a stats payload without stats

### PR DESCRIPTION
**Pull request by apiarian-datadog**
# What does this PR do?

Prevents seemingly unnecessary errors if a Serverless stats payload happens to not have any stats.

The payload definition in the go protobuf code seems to indicate that stats are allowed to be empty. And [that code](https://github.com/DataDog/datadog-agent/blob/main/pkg/trace/api/api.go#L525-L535) (`firstService` closure) is defensive about this possibility. Furthermore our `datadog.trace_agent.receiver.stats_buckets` metric has a minimum value of 0 so this apparently does happen and is handled as such by our systems.

# Motivation

The serverless agent is generating a bunch of noise when it receives these kinds of "empty" stats payloads from the node tracer.

# How to test the change?

Added a unit test to confirm that we are able to create stats payloads with these empty stats objects (and empty stats of stats... our naming is a bit hard to follow). Also checking that this does in fact work with an azure function and that we correctly don't have errors there.

### CODE REVIEW

**Comment by apiarian-datadog** [comment]
Before this change the agent had the following errors logged:
```
2024-08-02T16:56:55Z   [Information]   [2024-08-02T16:56:55Z INFO  datadog_trace_mini_agent::stats_processor] Recieved trace stats to process
2024-08-02T16:56:55Z   [Information]   [2024-08-02T16:56:55Z ERROR datadog_trace_mini_agent::http_utils] Error deserializing trace stats from request body: No stats in stats payload
```

With this change the agent shows the following:
```
2024-08-02T16:53:06Z   [Information]   [2024-08-02T16:53:05Z INFO  datadog_trace_mini_agent::stats_processor] Recieved trace stats to process
2024-08-02T16:53:06Z   [Information]   [2024-08-02T16:53:05Z INFO  datadog_trace_mini_agent::http_utils] Successfully buffered stats to be flushed.
2024-08-02T16:53:08Z   [Information]   [2024-08-02T16:53:07Z INFO  datadog_trace_mini_agent::stats_flusher] Flushing 1 stats
2024-08-02T16:53:08Z   [Information]   [2024-08-02T16:53:07Z INFO  datadog_trace_mini_agent::stats_flusher] Successfully flushed stats
```

**Review by duncanpharvey — APPROVED** [review]
LGTM!
